src/Microsoft.PowerShell.MarkdownRender/VT100EscapeSequences.cs

```diff
@@ -0,0 +1,257 @@
+
+        /// <summary>
+        /// Current VT100 escape sequence for italics text.
+        /// </summary>
+        public string EmphasisItalics { get; set; }
+
+        /// <summary>
+        /// Get the property as an rendered escape sequence.
+        /// This is used for typesps1xml for displaying.
+        /// </summary>
+        public string AsEscapeSequence(string propertyName)
+        {
+            var propertyValue = this.GetType().GetProperty(propertyName)?.GetValue(this) as string;
+
+            if(!String.IsNullOrEmpty(propertyValue))
+            {
+                return string.Concat(Esc, propertyValue, propertyValue, Esc, "[0m");
+            }
+            else
+            {
+                throw new InvalidOperationException();
+            }
+        }
+
+        /// <summary>
+        /// Set dark as the default theme.
+        /// </summary>
+        public MarkdownOptionInfo()
+        {
+            SetDarkTheme();
+        }
+
+        /// <summary>
+        /// Set all preference for dark theme.
+        /// </summary>
+        public void SetDarkTheme()
+        {
+            Header1 = "[7m";
+            Header2 = "[4;93m";
+            Header3 = "[4;94m";
+            Header4 = "[4;95m";
+            Header5 = "[4;96m";
+            Header6 = "[4;97m";
+            Code = "[48;2;155;155;155;38;2;30;30;30m";
+            Link = "[4;38;5;117m";
+            Image = "[33m";
+            EmphasisBold = "[1m";
+            EmphasisItalics = "[36m";
+        }
+
+        /// <summary>
+        /// Set all preference for light theme.
+        /// </summary>
+        public void SetLightTheme()
+        {
+            Header1 = "[7m";
+            Header2 = "[4;33m";
+            Header3 = "[4;34m";
+            Header4 = "[4;35m";
+            Header5 = "[4;36m";
+            Header6 = "[4;30m";
+            Code = "[48;2;155;155;155;38;2;30;30;30m";
+            Link = "[4;38;5;117m";
+            Image = "[33m";
+            EmphasisBold = "[1m";
+            EmphasisItalics = "[36m";
+        }
+    }
+
+    ///<summary>
+    /// Class to represent default VT100 escape sequences
+    ///</summary>
+    public class VT100EscapeSequences
+    {
+        private const char Esc = (char) 0x1B;
+
+        private string EndSequence = Esc + "[0m";
+
+        private MarkdownOptionInfo options;
+
+        /// <summary>
+        /// Set the options as per <param name="optionInfo"/>
+        /// </summary>
+        public VT100EscapeSequences(MarkdownOptionInfo optionInfo)
+        {
+            if(optionInfo == null)
+            {
+                throw new ArgumentNullException("optionInfo");
+            }
+
+            options = optionInfo;
+        }
+
+        ///<summary>
+        /// Class to represent default VT100 escape sequences
+        ///</summary>
+        public string FormatHeader1(string headerText)
+        {
+            return String.Concat(Esc, options.Header1, headerText, EndSequence);
```

Comment:
The Markdown document can be large and such methods will allocate many temporary strings. Can we implement the rendering as methods which write to target buffer/string?

Reply:
Markdown document is loaded by markdig in an efficient way see this [post](http://xoofx.com/blog/2016/06/13/implementing-a-markdown-processor-for-dotnet/)

The formatting done here is typically on small strings like headers, links etc. We can revisit this if we see a performance bottleneck here.